VSocks.md

```diff
@@ -0,0 +1,133 @@
+| | .-----.  .-----. | |
+| | |cont1|  |cont2| | |
+| | `-----'  `-----' | |
+| |       \   /      | |
+| |    .---------.   | |
+| |    |  agent  |   | |
+| |    `---------'   | |
+| |         |        | |
+| |    .-----------. | |
+| |POD |serial port| | |
+| `----|-----------|-' |
+|      |  socket   |   |
+|      `-----------'   |
+|           |          |
+|       .-------.      |
+|       | proxy |      |
+|       `-------'      |
+|           |          |
+|  .------./ \.------. |
+|  | shim |   | shim | |
+|  `------'   `------' |
+| Host                 |
+`----------------------'
+```
+
+The second way for processes to communicate, and the newer one, is by using
+[vsocks][4]. The communication through vsocks is not serial and can accept
+connections from multiple clients unlike serial ports devices, therefore no
+multiplexers ([kata-proxy][2] and [yamux][3]) are required. The following diagram
+shows how it's implemented in Kata Containers.
+
+
+### vsock communication diagram
+
+```
+.----------------------.
+| .------------------. |
+| | .-----.  .-----. | |
+| | |cont1|  |cont2| | |
+| | `-----'  `-----' | |
+| |       |   |      | |
+| |    .---------.   | |
+| |    |  agent  |   | |
+| |    `---------'   | |
+| |       |   |      | |
+| | POD .-------.    | |
+| `-----| vsock |----' |
+|       `-------'      |
+|         |   |        |
+|  .------.   .------. |
+|  | shim |   | shim | |
+|  `------'   `------' |
+| Host                 |
+`----------------------'
+```
+
+## System requirements
+
+Host linux kernel version must be greater than or equal to v4.8 and the
+`vhost_vsock` module must be loaded or built-in (CONFIG_VHOST_VSOCK=y). To
+load the module run the following command:
+
+```
+$ sudo modprobe -i vhost_vsock
+```
+
+Kata Containers version must be greater than or equal to 1.2.0 and `use_vsock`
+must be `true` in the runtime [configuration file][1].
+
+## Advantages of using vsocks
+
+### High density
+
+Having no proxy running in the host multiplexing the connections between virtual
+machine and host processes saves about 4.5MB of RAM less per [POD][5].
+When we talk about density each kilobyte matters and it might be the decisive
+factor between run another container or not. For example if you have 500
+PODs running in a server, the same amount of [kata-proxy][2] processes will be
+running and consuming for around 2250MB of RAM. Before making the decision not to
+use vsocks, you should ask yourself, how many containers more can run with the
+memory RAM consumed by the kata-proxies?.
+
+### Reliability
+
+Since [kata-proxy][2] is in charge of multiplexing the connections between
+virtual machine and host processes, if it dies all connections get broken. For
+example if you have a [POD][5] with 10 containers running, if kata-proxy dies
+your 10 containers would be like aliens, they are there but it's impossible to
+contact them.
+As was mentioned in the [Introduction](#introduction), when vsocks are used to
```

Comment:
Line 121-124 - How about "Since communication via vsocks is direct, the only way to lose communication with the virtual machine is if the VM itself or the [shim][6] dies."